src/odemis/acq/stream/_sync.py

```diff
@@ -1833,17 +2068,31 @@ def _runAcquisition(self, future):
                         if self._acq_state == CANCELLED:
                             raise CancelledError()
 
+                self._acq_data = [[] for _ in self._streams]  # delete acq_data to use less RAM
             # Done!
             with self._acq_lock:
                 if self._acq_state == CANCELLED:
                     raise CancelledError()
                 self._acq_state = FINISHED
+            self._current_scan_area = None
 
-            # TODO: ideally, we would directly assemble the _acq_data into the final
-            # raw, which would avoid temporarily holding twice all the data.
+            # broadcast all the self._live_data into self._raw and do post-processing
+            for stream_idx, das in enumerate(self._live_data):
+                if stream_idx == 0 and len(das) == 0:
+                    # It's OK to not have data for the SEM stream (e.g. Monochromator)
+                    continue
+                self._assembleFinalData(stream_idx, das)
 
-            for n, das in enumerate(self._acq_data):
-                self._onCompletedData(n, das)
+                if isinstance(self._streams[stream_idx], MonochromatorSettingsStream):
+                    # The Monochromator stream uses a chronogram view, which is not compatible with the RGB spatial image
+                    continue
+                elif stream_idx >= 1:
+                    # Only update the CL stream
+                    self._streams[1].raw = [self._live_data[1][-1]]  # For GetBoundingBox()
+                    rgbim = self._projectXY2RGB(self._live_data[1][-1])
+                    self._streams[1].image.value = rgbim
+                    #TODO K.K. what is better? Above or below?
+                    # self._streams[stream_idx]._onNewData(self._streams[stream_idx]._dataflow, self._raw[-1])
```

Comment:
@pieleric I am not sure if my implementation is actually that much better, which one do you prefer?

Reply:
after this I will remove all the _Oncompletedata, remove all the unnecacerry changes etc. and then this should be ready to be merged.

Reply:
I think that if you end-up still needing this "isinstance()" (because you cannot call `updateImage()` directly, then just go for the shorter version, with `onNewData()`.
BTW, I'd suggest to put this block concerning the image update into a try/except block, and only log the exceptions, because we don't want any issue here to cause a complete failure of the acquisition.